pkg/cmd/cli/cmd/importimage.go

```diff
@@ -320,8 +320,12 @@ func (o *ImportImageOptions) createImageImport() (*imageapi.ImageStream, *imagea
 			}
 
 			if ok {
+				// disallow re-importing anything other than DockerImage
+				if existing.From != nil && existing.From.Kind != "DockerImage" {
+					return nil, nil, fmt.Errorf("tag %q points to existing %s %q, it cannot be re-imported", tag, existing.From.Kind, existing.From.Name)
+				}
 				// disallow changing an existing tag
-				if existing.From == nil || existing.From.Kind != "DockerImage" {
+				if existing.From == nil {
 					return nil, nil, fmt.Errorf("tag %q already exists - you must use the 'tag' command if you want to change the source to %q", tag, from)
```

Comment:
Quoting the error message from the mentioned bz:

> error: tag "latest" already exists - you must use the 'tag' command if you want to change the source to ""

Since the `from` may be unset either strip the `to %q` suffix or make it appear depending on `from`.